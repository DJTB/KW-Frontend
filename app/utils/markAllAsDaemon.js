/* eslint-disable no-param-reassign */

/**
 * Marks sagas as daemons so they will only run once in react-router onEntry handler
 * @param  {Array} list of saga functions
 * @return {Array} new list with sagas marked as daemons
 */
function markAllAsDaemon(sagas = []) {
  return sagas.slice().map((saga) => {
    saga.isDaemon = true;
    return saga;
  });
}

export default markAllAsDaemon;
