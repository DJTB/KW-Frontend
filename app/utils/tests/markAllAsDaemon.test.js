import markAllAsDaemon from '../markAllAsDaemon';

describe('markAllAsDaemon()', () => {
  it('should match snapshot', () => {
    expect(markAllAsDaemon([{}, {}])).toMatchSnapshot();
  });
});
