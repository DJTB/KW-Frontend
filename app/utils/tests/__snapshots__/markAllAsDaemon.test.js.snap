// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markAllAsDaemon() should match snapshot 1`] = `
Array [
  Object {
    "isDaemon": true,
  },
  Object {
    "isDaemon": true,
  },
]
`;
